src/modules/map/components/cesium/CesiumWMSLayer.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+    <div>
+        <slot />
+    </div>
+</template>
+
+<script>
+import { CoordinateSystem, WEBMERCATOR } from '@/utils/coordinateSystems'
+import { ImageryLayer, Rectangle, WebMapServiceImageryProvider } from 'cesium'
+import { WGS84_EXTENT } from '@/config'
+import { mapState } from 'vuex'
+import { YEAR_TO_DESCRIBE_ALL_OR_CURRENT_DATA } from '@/api/layers/LayerTimeConfigEntry.class'
+import GeoAdminWMSLayer from '@/api/layers/GeoAdminWMSLayer.class'
+import addImageryLayerMixins from '@/modules/map/components/cesium/utils/addImageryLayer-mixins'
+
+// todo should we adapt LOD to resolution?
+const MAXIMUM_LEVEL_OF_DETAILS = 18
+
+export default {
+    mixins: [addImageryLayerMixins],
+    props: {
+        wmsLayerConfig: {
+            type: [GeoAdminWMSLayer],
+            required: true,
+        },
+        previewYear: {
+            type: Number,
+            default: null,
+        },
+        projection: {
+            type: CoordinateSystem,
+            default: WEBMERCATOR.epsg,
+        },
+        zIndex: {
+            type: Number,
+            default: -1,
+        },
+    },
+    computed: {
+        ...mapState({
+            currentLang: (state) => state.i18n.lang,
+        }),
+        layerId() {
+            return this.wmsLayerConfig.geoAdminID || this.wmsLayerConfig.externalLayerId
+        },
+        opacity() {
+            return this.wmsLayerConfig.opacity || 1.0
+        },
+        wmsVersion() {
+            return this.wmsLayerConfig.wmsVersion || '1.3.0'
+        },
+        format() {
+            return this.wmsLayerConfig.format || 'png'
+        },
+        url() {
+            return this.wmsLayerConfig.getURL(this.previewYear, this.projection.epsgNumber)
+        },
+        timestamp() {
+            if (this.wmsLayerConfig.timeConfig) {
+                // if there is a preview year set, we search for the matching timestamp
+                if (this.previewYear) {
+                    const matchingTimeEntry = this.wmsLayerConfig.getTimeEntryForYear(
+                        this.previewYear
+                    )
+                    if (matchingTimeEntry) {
+                        return matchingTimeEntry.timestamp
+                    }
+                }
+                // if a time entry is defined, and is different from 'all'
+                // (no need to pass 'all' to our WMS, that's the default timestamp used under the hood)
+                if (
+                    this.wmsLayerConfig.timeConfig.currentYear !==
+                    YEAR_TO_DESCRIBE_ALL_OR_CURRENT_DATA
+                ) {
+                    return this.wmsLayerConfig.timeConfig.currentTimestamp
+                }
+            }
+            return ''
```

Comment:
As this is essentially a copy past of `src/modules/map/components/openlayers/OpenLayersWMSLayer.vue` from line `65...86`, I would prefer that we move this logic in a small simple `wmsLayerUtils.js` in the `src/utils/` folder, or something in this direction.
If we want to change in the future the logic behind previewYear, it would not be good to have duplication of code such as this one.

Reply:
moved